=== modified file 'Makefile.in'
--- old/Makefile.in	2013-07-23 22:55:38 +0000
+++ new/Makefile.in	2013-08-07 22:03:11 +0000
@@ -318,7 +318,7 @@
 
 # Convert MSYS-style /x/foo or Windows-style x:\foo file names
 # into x:/foo that Windows can grok.
-msys_to_w32=sed -e 's,\\\\,/,g' -e 's,^/\([A-Za-z]\)/,\1:/,'
+msys_to_w32=sed -e 's,\\\\,/,g' -e 's,^/cygdrive/\([A-Za-z]\)/,\1:/,' -e 's,^/\([A-Za-z]\)/,\1:/,'
 
 # Transform directory search path and its components.  Original can
 # be MSYS or Windows style.  Set path separator to ";", directory
@@ -326,6 +326,7 @@
 # Remove empty path components and escape semicolons.
 msys_lisppath_to_w32=sed -e 's,\\\\,/,g' 			\
 	-e 's,\(^\|[:;]\)\([A-Za-z]\):/,\1/\2/,g'		\
+	-e 's/:/;/g' -e 's,\(^\|;\)/cygdrive/\([A-Za-z]\)/,\1\2:/,g'	\
 	-e 's/:/;/g' -e 's,\(^\|;\)/\([A-Za-z]\)/,\1\2:/,g'	\
 	-e 's/;\+/;/g' -e 's/^;//' -e 's/;$$//' -e 's/;/\\\\;/g'
 

=== modified file 'configure.ac'
--- old/configure.ac	2013-08-04 16:56:56 +0000
+++ new/configure.ac	2013-08-07 22:03:11 +0000
@@ -594,6 +594,18 @@
     esac
   ;;
 
+  ## x86_64 machines where we don't care about the manufacturer.
+  x86_64-*-* )
+    case "${canonical}" in
+      *-mingw32 )
+                opsys=mingw32
+                # MinGW overrides and adds some system headers in nt/inc.
+                GCC_TEST_OPTIONS="-I $srcdir/nt/inc"
+                ;;
+      ## Otherwise, we'll fall through to the generic opsys code at the bottom.
+    esac
+  ;;
+
   * )
     unported=yes
   ;;
@@ -1126,7 +1138,7 @@
 ## additional optimization.  --nils@exp-math.uni-essen.de
 test "$opsys" = "aix4.2" && test "x$GCC" != "xyes" && \
   C_SWITCH_SYSTEM="-ma -qmaxmem=4000"
-test "$opsys" = "mingw32" && C_SWITCH_SYSTEM="-mtune=pentium4"
+#test "$opsys" = "mingw32" && C_SWITCH_SYSTEM="-mtune=pentium4"
 ## gnu-linux might need -D_BSD_SOURCE on old libc5 systems.
 ## It is redundant in glibc2, since we define _GNU_SOURCE.
 AC_SUBST(C_SWITCH_SYSTEM)
@@ -3152,6 +3164,11 @@
       LIBXML2_CFLAGS="-I/usr/include/libxml2"
     fi
   fi
+  if test "${HAVE_LIBXML2}" != "yes" -a "$opsys" = "mingw32"; then
+  AC_CHECK_HEADER(libxml/HTMLparser.h,
+  [AC_CHECK_DECL(HTML_PARSE_RECOVER, HAVE_LIBXML2=yes, ,
+  [#include <libxml/HTMLparser.h>])])
+  fi
   if test "${HAVE_LIBXML2}" = "yes"; then
     if test "${opsys}" != "mingw32"; then
       LIBS="$LIBXML2_LIBS $LIBS"

=== modified file 'lib-src/Makefile.in'
--- old/lib-src/Makefile.in	2013-07-23 22:55:38 +0000
+++ new/lib-src/Makefile.in	2013-08-07 22:03:11 +0000
@@ -302,13 +302,14 @@
 TAGS: etags${EXEEXT}
 	etags *.[ch]
 
+msys_to_w32=sed -e 's,\\\\,/,g' -e 's,^/cygdrive/\([A-Za-z]\)/,\1:/,' -e 's,^/\([A-Za-z]\)/,\1:/,'
 ## This verifies that the non-ASCII characters in the file \`testfile\'
 ## have not been clobbered by whatever means were used to copy and
 ## distribute Emacs.  If they were clobbered, all the .elc files were
 ## clobbered too.
 test-distrib${EXEEXT}: ${srcdir}/test-distrib.c
 	$(CC) ${ALL_CFLAGS} -o test-distrib${EXEEXT} ${srcdir}/test-distrib.c
-	./test-distrib ${srcdir}/testfile
+	./test-distrib `echo ${srcdir}/testfile | ${msys_to_w32}`
 
 ../lib/libgnu.a: $(config_h)
 	cd ../lib && $(MAKE) libgnu.a

=== modified file 'nt/inc/sys/time.h'
--- old/nt/inc/sys/time.h	2013-06-03 17:15:44 +0000
+++ new/nt/inc/sys/time.h	2013-08-07 22:03:11 +0000
@@ -14,8 +14,13 @@
 #define _TIMEVAL_DEFINED
 struct timeval
 {
+#ifndef _WIN64
   long		tv_sec;		/* seconds */
   long		tv_usec;	/* microseconds */
+#else
+  unsigned long long	tv_sec;		/* seconds */
+  long int long		tv_usec;	/* microseconds */
+#endif
 };
 #define timerisset(tvp)  ((tvp)->tv_sec || (tvp)->tv_usec)
 #define timercmp(tvp, uvp, cmp) \

=== modified file 'src/Makefile.in'
--- old/src/Makefile.in	2013-07-23 23:00:43 +0000
+++ new/src/Makefile.in	2013-08-07 22:03:11 +0000
@@ -454,6 +454,7 @@
 	  ln emacs$(EXEEXT) bootstrap-emacs$(EXEEXT); \
 	fi
 
+msys_to_w32=sed -e 's,\\\\,/,g' -e 's,^/cygdrive/\([A-Za-z]\)/,\1:/,' -e 's,^/\([A-Za-z]\)/,\1:/,'
 ## We run make-docfile twice because the command line may get too long
 ## on some systems.  The sed command operating on lisp.mk also reduces
 ## the length of the command line.  Unfortunately, no-one has any idea
@@ -471,8 +472,8 @@
 $(etc)/DOC: $(libsrc)/make-docfile$(EXEEXT) $(obj) $(lisp)
 	$(MKDIR_P) $(etc)
 	-rm -f $(etc)/DOC
-	$(libsrc)/make-docfile -d $(srcdir) $(SOME_MACHINE_OBJECTS) $(obj) > $(etc)/DOC
-	$(libsrc)/make-docfile -a $(etc)/DOC -d $(lispsource) `sed -n -e 's| \\\\||' -e 's|^[ 	]*$$(lispsource)/||p' $(srcdir)/lisp.mk`
+	$(libsrc)/make-docfile -d `echo $(srcdir)|${msys_to_w32}` $(SOME_MACHINE_OBJECTS) $(obj) > $(etc)/DOC
+	$(libsrc)/make-docfile -a $(etc)/DOC -d `echo $(lispsource)|${msys_to_w32}` `sed -n -e 's| \\\\||' -e 's|^[ 	]*$$(lispsource)/||p' $(srcdir)/lisp.mk`
 
 $(libsrc)/make-docfile$(EXEEXT):
 	cd $(libsrc); $(MAKE) $(MFLAGS) make-docfile$(EXEEXT)
@@ -489,7 +490,7 @@
 GLOBAL_SOURCES = $(base_obj:.o=.c) $(NS_OBJC_OBJ:.o=.m)
 
 gl-stamp: $(libsrc)/make-docfile$(EXEEXT) $(GLOBAL_SOURCES)
-	$(libsrc)/make-docfile -d $(srcdir) -g $(obj) > gl.tmp
+	$(libsrc)/make-docfile -d `echo $(srcdir) | ${msys_to_w32}` -g $(obj) > gl.tmp
 	$(srcdir)/../build-aux/move-if-change gl.tmp globals.h
 	echo timestamp > $@
 

=== modified file 'src/image.c'
--- old/src/image.c	2013-08-03 03:29:03 +0000
+++ new/src/image.c	2013-08-07 22:03:11 +0000
@@ -5599,7 +5599,7 @@
    substitute may munge the signal mask, but that should be OK here.
    MinGW (MS-Windows) uses _setjmp and defines setjmp to _setjmp in
    the system header setjmp.h; don't mess up that.  */
-#ifndef HAVE__SETJMP
+#if 0 //#ifndef HAVE__SETJMP
 # define _setjmp(j) setjmp (j)
 # define _longjmp longjmp
 #endif
@@ -5797,7 +5797,7 @@
 
   /* Set error jump-back.  We come back here when the PNG library
      detects an error.  */
-  if (_setjmp (PNG_JMPBUF (png_ptr)))
+  if (sys_setjmp (PNG_JMPBUF (png_ptr)))
     {
     error:
       if (c->png_ptr)

=== modified file 'src/w32heap.c'
--- old/src/w32heap.c	2013-01-22 13:30:17 +0000
+++ new/src/w32heap.c	2013-08-07 22:03:11 +0000
@@ -96,7 +96,7 @@
 allocate_heap (void)
 {
 #ifdef _WIN64
-  size_t size = 0x4000000000i64; /* start by asking for 32GB */
+  size_t size = 0x4000000000ull; /* start by asking for 32GB */
 #else
   /* We used to start with 2GB here, but on Windows 7 that would leave
      too little room in the address space for threads started by

=== modified file 'src/w32proc.c'
--- old/src/w32proc.c	2013-02-25 17:36:03 +0000
+++ new/src/w32proc.c	2013-08-07 22:03:11 +0000
@@ -2251,7 +2251,7 @@
 
 /* Emulate 'kill', but only for other processes.  */
 int
-sys_kill (pid_t pid, int sig)
+sys_kill (int pid, int sig)
 {
   child_process *cp;
   HANDLE proc_hand;

